cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/services/AppFabricServer.java

```diff
@@ -83,12 +81,6 @@ public class AppFabricServer extends AbstractIdleService {
   private final ApplicationLifecycleService applicationLifecycleService;
   private final Set<String> servicesNames;
   private final Set<String> handlerHookNames;
-  private final AuditLogSubscriberService auditLogSubscriberService;
-  private final ProgramNotificationSubscriberService programNotificationSubscriberService;
-  private final ProgramStopSubscriberService programStopSubscriberService;
-  private final RunRecordCorrectorService runRecordCorrectorService;
-  private final RunDataTimeToLiveService runDataTimeToLiveService;
-  private final ProgramRunStatusMonitorService programRunStatusMonitorService;
   private final RunRecordMonitorService runRecordCounterService;
```

Comment:
This guy is not designed to run in multiple instances. It has internal tracker for "number of concurrent starts". Our counters will be wrong if we start it everywhere, e.g. removeRequest is mostly called by ProgramNotificationSubscriberServicem while new requests are added by the handler. We need to make this mechanism distributed. @masoud-io FYI

Reply:
As discussed, in this PR I have moved the HTTP handlers for program, app, workflow to processors.

They can be moved back to the Appfabric server after the issue due to in-memory cache is fixed. Created Jira for follow up PR: [CDAP-21112](https://cdap.atlassian.net/browse/CDAP-21112) `Move HTTP handler from Appfabric processor to server after fixing ProgramRuntimeService and RunRecordMonitorService`